lib/matplotlib/bezier.py

```diff
@@ -168,27 +176,114 @@ def find_bezier_t_intersecting_with_closedpath(
 
 class BezierSegment:
     """
-    A D-dimensional Bezier segment.
+    A d-dimensional Bezier segment.
 
     Parameters
     ----------
-    control_points : (N, D) array
+    control_points : (N, d) array
         Location of the *N* control points.
     """
 
     def __init__(self, control_points):
-        n = len(control_points)
-        self._orders = np.arange(n)
-        coeff = [math.factorial(n - 1)
-                 // (math.factorial(i) * math.factorial(n - 1 - i))
-                 for i in range(n)]
-        self._px = np.asarray(control_points).T * coeff
+        self._cpoints = np.asarray(control_points)
+        self._N, self._d = self._cpoints.shape
+        self._orders = np.arange(self._N)
+        coeff = [math.factorial(self._N - 1)
+                 // (math.factorial(i) * math.factorial(self._N - 1 - i))
+                 for i in range(self._N)]
+        self._px = self._cpoints.T * coeff
+
+    def __call__(self, t):
```

Comment:
Totally valid. This seems to be an artifact of splitting up my work into multiple PRs. In the next PR in the stack, a vectorized version of point_at_t becomes necessary, and so in order to prevent an API change, I implement `__call__` to work vectorized, and have `point_at_t` return `tuple(self(t))`. 

I'll remove this here and wait for the next PR where it's necessary.

Reply:
This has been moved back here, since I now use it to implement the polynomial coefficients calculation in a vectorized way.